### PR TITLE
:recycle: refactor: merge some external middlewares to core

### PR DESCRIPTION
**Pull request by efectn**
## Description
This PR merges adaptor, helmet, keyauth, redirect, rewrite middlewares to reduce maintenance. 

- [x] Fix linting issues.
- [x] Add descriptions of some middlewares.
- [x] Update README.

## Type of change
- [x] Refactoring
- [x] This change requires a documentation update

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation - /docs/ directory for https://docs.gofiber.io/
- [x] New and existing unit tests pass locally with my changes

## Commit formatting:
Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

File: middleware/keyauth/keyauth_test.go [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
```
Comment (gaby) [inline middleware/keyauth/keyauth_test.go:14]: Should this use these utils or the utils repo?

File: middleware/redirect/redirect_test.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
+//nolint:bodyclose // Much easier to just ignore memory leaks in tests
+package redirect
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
```
Comment (gaby) [inline middleware/redirect/redirect_test.go:10]: Same question here

File: middleware/rewrite/rewrite_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
```
Comment (gaby) [inline middleware/rewrite/rewrite_test.go:12]: Same here

File: middleware/adaptor/adaptor.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+import (
+	"io"
+	"net"
+	"net/http"
+	"reflect"
+	"unsafe"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
```
Comment (gaby) [inline middleware/adaptor/adaptor.go:11]: Here too

**Review by gaby — CHANGES_REQUESTED** [review]

File: middleware/keyauth/keyauth_test.go (hunk 1 above)
Comment (efectn) [reply middleware/keyauth/keyauth_test.go:14]: We will use utils repo in v3

**Comment by ReneWerner87** [comment]
thank you, looks good at first glance, but still need to look more closely

will merge it on the weekend

**Comment by ReneWerner87** [comment]
I will merge it and then look at it in the master, in case of errors we have to open follow-up PR's.

**Pull request merged**